### PR TITLE
[develop-0.1] chore: backport CI changes from develop

### PR DESCRIPTION
**Pull request by steveeJ**
### Summary

this synchronizes all CI config files with the develop-0.1 branch.

i used the approach of running `git checkout develop -- [relevant paths...]` and then `git commit` as opposed to cherry-picking the relevant commits. 

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [x] debug failing tests
- [x] double-check the test feature sets ares appropriate @maackle 
- [ ] double-check the ignored tests ares appropriate @maackle  
- [ ] release dry-run
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: holonix/README.md [hunk 1]
```diff
@@ -1,3 +1,59 @@
...
+              packages = with pkgs; [
+                  # more packages go here
+              ];
+            };
+          };
+      };
+}
+```
+
+this exmaple would translate to the following CLI invocatin
```
Comment (maackle) [inline holonix/README.md:52]:
```suggestion
this example would translate to the following CLI invocation
```

**Review by maackle — APPROVED** [review]
I can't review all 3.6k lines, but I checked that the features seem to be still correctly in place. I'm not sure what you meant by checking that ignored tests are appropriate.

**Comment by jost-s** [comment]
@steveeJ 
I've rebased the backport branch on this branch as a test. Circle CI jobs aren't running, as intended, but there are no other pipeline runs either. Shouldn't the Github action pipelines be run? https://github.com/holochain/holochain/pull/2134

**Comment by steveeJ** [comment]
> @steveeJ I've rebased the backport branch on this branch as a test. Circle CI jobs aren't running, as intended, but there are no other pipeline runs either. Shouldn't the Github action pipelines be run? #2134

my guess is that github was confused by the change  of target branch.

image


maybe try pushing there again, even if it's just an empty commit (`git commit --allow-empty`)

**Review by jost-s — APPROVED** [review]
Yes, after the push the pipelines ran. However, there are some errors related to code. That means the backport branch will show those errors too once this CI backport branch is merged.

The errors don't seem related to this branch, so I'll approve as a best guess =)

**Pull request merged**